packages/hardhat-ethers/test/hardhat-ethers-provider.ts

```diff
@@ -117,12 +119,83 @@ describe("hardhat ethers provider", function () {
     assert.strictEqual(network.chainId, 31337n);
   });
 
-  it("should return fee data", async function () {
-    const feeData = await this.env.ethers.provider.getFeeData();
+  describe("getFeeData", function () {
+    it("should return fee data", async function () {
+      const feeData = await this.env.ethers.provider.getFeeData();
 
-    assert.typeOf(feeData.gasPrice, "bigint");
-    assert.typeOf(feeData.maxFeePerGas, "bigint");
-    assert.typeOf(feeData.maxPriorityFeePerGas, "bigint");
+      assert.typeOf(feeData.gasPrice, "bigint");
+      assert.typeOf(feeData.maxFeePerGas, "bigint");
+      assert.typeOf(feeData.maxPriorityFeePerGas, "bigint");
+    });
+
+    // This helper overrides the send method of an EthereumProvider to allow
+    // altering the default Hardhat node's reported results.
+    function overrideSendOn(
+      provider: EthereumProvider,
+      sendOveride: (method: string, params?: any[] | undefined) => Promise<any>
+    ) {
+      return new Proxy(provider, {
```

Comment:
My knowledge of providers is not high enough to know if there is a better pattern for overriding the RPC mechanism of the underlying Ethereum provider.
If there is a cleaner approach, let me know.

Reply:
I think we have a mock provider object somewhere, but let's move fwd with this.